packages/contracts-bedrock/contracts/legacy/L1ChugSplashProxy.sol

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+/**
+ * @title iL1ChugSplashDeployer
+ */
+interface iL1ChugSplashDeployer {
+    function isUpgrading() external view returns (bool);
+}
+
+/**
+ * @title L1ChugSplashProxy
+ * @dev Basic ChugSplash proxy contract for L1. Very close to being a normal proxy but has added
+ * functions `setCode` and `setStorage` for changing the code or storage of the contract. Nifty!
+ *
+ * Note for future developers: do NOT make anything in this contract 'public' unless you know what
+ * you're doing. Anything public can potentially have a function signature that conflicts with a
+ * signature attached to the implementation contract. Public functions SHOULD always have the
+ * 'proxyCallIfNotOwner' modifier unless there's some *really* good reason not to have that
+ * modifier. And there almost certainly is not a good reason to not have that modifier. Beware!
+ */
+contract L1ChugSplashProxy {
+    /*************
+     * Constants *
+     *************/
+
+    // "Magic" prefix. When prepended to some arbitrary bytecode and used to create a contract, the
+    // appended bytecode will be deployed as given.
+    bytes13 internal constant DEPLOY_CODE_PREFIX = 0x600D380380600D6000396000f3;
+
+    // bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)
+    bytes32 internal constant IMPLEMENTATION_KEY =
+        0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+
+    // bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)
+    bytes32 internal constant OWNER_KEY =
+        0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _owner Address of the initial contract owner.
+     */
+    constructor(address _owner) {
+        _setOwner(_owner);
```

Comment:
similar to my previous comment, I think we should just say `_admin` here and avoid the term `owner` in general as the implementation contracts will typically have an owner.

Reply:
I would prefer to have 0 diff with the previous contract

Reply:
Oh good point. I was actually meant this comment in the context of the new Proxy. 


https://github.com/ethereum-optimism/optimism/blob/22b8cf19aa5715177aa553a6fe3878a4c2252c5c/packages/contracts-bedrock/contracts/universal/Proxy.sol#L51

Reply:
That'd also be more consistent with the usage: https://github.com/ethereum-optimism/optimism/blob/22b8cf19aa5715177aa553a6fe3878a4c2252c5c/packages/contracts-bedrock/contracts/universal/ProxyAdmin.sol#L204